test/modules/print/DailyPrintView.test.js

```diff
@@ -14,3 +14,348 @@
+    });
+
+    it('should set default properties as provided by constructor args', () => {
+      expect(Renderer.doc).to.eql(doc);
+      expect(Renderer.data).to.eql(data);
+
+      const overrideOpts = [
+        'debug',
+        'dpi',
+        'margins',
+        'defaultFontSize',
+        'footerFontSize',
+        'headerFontSize',
+        'summaryHeaderFontSize',
+        'bgPrefs',
+        'timePrefs',
+        'width',
+        'height',
+        'chartsPerPage',
+        'numDays',
+        'patient',
+      ];
+
+      _.each(overrideOpts, opt => {
+        expect(Renderer[opt]).to.equal(opts[opt]);
+      });
+
+      expect(Renderer.bgUnits).to.equal(opts.bgPrefs.bgUnits);
+      expect(Renderer.bgBounds).to.equal(opts.bgPrefs.bgBounds);
+      expect(Renderer.timezone).to.equal(getTimezoneFromTimePrefs(opts.timePrefs));
+    });
+
+    it('should set it\'s own required initial instance properties', () => {
+      const requiredProps = [
+        { prop: 'font', type: 'string' },
+        { prop: 'boldFont', type: 'string' },
+        { prop: 'bgAxisFontSize', type: 'number' },
+        { prop: 'carbsFontSize', type: 'number' },
+        { prop: 'bolusWidth', type: 'number' },
+        { prop: 'carbRadius', type: 'number' },
+        { prop: 'cbgRadius', type: 'number' },
+        { prop: 'extendedLineThickness', type: 'number' },
+        { prop: 'interruptedLineThickness', type: 'number' },
+        { prop: 'smbgRadius', type: 'number' },
+        { prop: 'triangleHeight', type: 'number' },
+        { prop: 'startingPageIndex', type: 'number', value: opts.startingPageIndex || 0 },
+        { prop: 'totalPages', type: 'number', value: 0 },
+        { prop: 'chartsPlaced', type: 'number', value: 0 },
+        { prop: 'chartIndex', type: 'number', value: 0 },
+        { prop: 'colors', type: 'object' },
+        { prop: 'gapBtwnSummaryAndChartAsPercentage', type: 'number' },
+        { prop: 'rightEdge', type: 'number', value: Renderer.margins.left + Renderer.width },
+        { prop: 'bottomEdge', type: 'number', value: Renderer.margins.top + Renderer.height },
+        { prop: 'patientInfoBox', type: 'object', value: {
+          width: 0,
+          height: 0,
+        } },
+        { prop: 'chartArea', type: 'object' },
+        { prop: 'initialChartArea', type: 'object', value: {
+          bottomEdge: opts.margins.top + opts.height,
+          leftEdge: opts.margins.left +
+          (opts.summaryWidthAsPercentage + Renderer.gapBtwnSummaryAndChartAsPercentage) *
+          Renderer.width,
+          topEdge: opts.margins.top,
+          width: Renderer.rightEdge - (opts.margins.left +
+          (opts.summaryWidthAsPercentage + Renderer.gapBtwnSummaryAndChartAsPercentage) *
+          Renderer.width),
+        } },
+      ];
+
+      _.each(requiredProps, item => {
+        expect(Renderer[item.prop]).to.be.a(item.type);
+        item.value && expect(Renderer[item.prop]).to.eql(item.value);
```

Comment:
Won't this `item.value` return false for numeric `0` and short circuit the `expect`?